controllers/etcdpeer_controller_test.go

```diff
@@ -1,62 +1,52 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
 package controllers
 
 import (
-	"context"
+	"github.com/avast/retry-go"
+	"testing"
 	"time"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
 	appsv1 "k8s.io/api/apps/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	etcdv1alpha1 "github.com/improbable-eng/etcd-cluster-operator/api/v1alpha1"
 )
 
-var _ = Describe("Etcd peer controller", func() {
-	ctx := context.Background()
-	SetupTest(ctx)
-
-	Context("ReplicaSet creation", func() {
-		It("Should create a ReplicaSet if one does not exist already", func() {
-
-			By("Creating a EtcdPeer")
-			etcdPeer := etcdv1alpha1.EtcdPeer{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels:      make(map[string]string),
-					Annotations: make(map[string]string),
-					Name:        "bees",
-					Namespace:   "default",
-				},
-			}
-
-			err := k8sClient.Create(ctx, &etcdPeer)
-			Expect(err).NotTo(HaveOccurred(), "failed to create EtcdPeer resource")
-
-			By("The controller having created a ReplicaSet")
-			replicaSet := &appsv1.ReplicaSet{}
-			Eventually(
-				getResourceFunc(
-					ctx,
-					client.ObjectKey{
-						// Same name and namespace as the EtcdPeer above
-						Name:      etcdPeer.Name,
-						Namespace: etcdPeer.Namespace,
-					},
-					replicaSet,
-				),
-				// Timeout
-				time.Second*5,
-				// Polling Interval
-				time.Millisecond*500,
-			).Should(BeNil())
-		})
-	})
-})
+func (s *controllerSuite) testPeerController(t *testing.T) {
+	t.Run("TestPeerController_OnCreation_CreatesReplicaSet", func(t *testing.T) {
+		teardownFunc := s.setupTest(t)
+		defer teardownFunc()
+
+		etcdPeer := &etcdv1alpha1.EtcdPeer{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels:      make(map[string]string),
+				Annotations: make(map[string]string),
+				Name:        "bees",
+				Namespace:   "default",
+			},
+		}
+
+		err := s.k8sClient.Create(s.ctx, etcdPeer)
+		require.NoError(t, err, "failed to create EtcdPeer resource")
+
+		replicaSet := &appsv1.ReplicaSet{}
+
+		err = retry.Do(
+			func() error {
+				return s.k8sClient.Get(s.ctx, client.ObjectKey{
+					// Same name and namespace as the EtcdPeer above
+					Name:      etcdPeer.Name,
+					Namespace: etcdPeer.Namespace,
+				}, replicaSet)
+			},
+			retry.Attempts(10),
+			retry.Delay(time.Millisecond*500),
+			retry.DelayType(retry.FixedDelay),
+		)
```

Comment:
I'm not too familiar with testify/require, but is this the equivalent of `Eventually` in Gomega?
Out of interest, what would a `Consistently` block look like? These are pretty useful when testing (as eventually consistent systems are sneaky and can be a PITA to assert something actually finished reconciling at state X 😅)

Reply:
this package is actually [separate](https://github.com/avast/retry-go) from the testify suite. i'm not aware of an alternative, but an equivalent should be trivial to implement, e.g. https://play.golang.com/p/XTVY5hcEoSP